cogs/audio.py

```diff
@@ -1491,26 +1491,26 @@ def _clean_url(self, url):
 
         # Checking already connected, will join if not
 
-        try:
-            self.has_connect_perm(author, server)
-        except AuthorNotConnected:
-            await self.bot.say("You must join a voice channel before I can"
-                               " play anything.")
-            return
-        except UnauthorizedConnect:
-            await self.bot.say("I don't have permissions to join your"
-                               " voice channel.")
-            return
-        except UnauthorizedSpeak:
-            await self.bot.say("I don't have permissions to speak in your"
-                               " voice channel.")
-            return
-        except ChannelUserLimit:
-            await self.bot.say("Your voice channel is full.")
-            return
-
         if not self.voice_connected(server):
-            await self._join_voice_channel(voice_channel)
+            try:
+                self.has_connect_perm(author, server)
+            except AuthorNotConnected:
+                await self.bot.say("You must join a voice channel before I can"
+                                   " play anything.")
+                return
+            except UnauthorizedConnect:
+                await self.bot.say("I don't have permissions to join your"
+                                   " voice channel.")
+                return
+            except UnauthorizedSpeak:
+                await self.bot.say("I don't have permissions to speak in your"
+                                   " voice channel.")
+                return
+            except ChannelUserLimit:
+                await self.bot.say("Your voice channel is full.")
+                return
+            else:
+                await self._join_voice_channel(voice_channel)
         else:  # We are connected but not to the right channel
```

Comment:
Having the else block handled like this means the above permission checking never gets called if you want to bot to switch channels.